lib/battle/BattleInfo.cpp

```diff
@@ -183,6 +183,44 @@ struct RangeGenerator
 	std::function<int()> myRand;
 };
 
+void BattleInfo::adjustOppositeBonuses(CBonusSystemNode * node, PlayerColor ownerColor)
+{
+	auto & bonusList = node->getExportedBonusList();
+	if(bonusList.empty())
+		return;
+
+	for(const auto & bonus : bonusList)
+	{
+		if(bonus->effectRange != Bonus::ONLY_ENEMY_ARMY)
+			continue;
+
+		static_cast<OppositeSideLimiter *>(bonus->limiter.get())->owner = ownerColor;

```

Comment:
We should not crash in case of no limiter here.

Reply:
Case of no limiter here is a weird error. So, check and log are added